internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -78,76 +76,74 @@ type tektonTasks struct {
 
 var _ operands.Operand = &tektonTasks{}
 
-func New(bundle *tektonbundle.Bundle) *tektonTasks {
-	tt := &tektonTasks{
+func New(bundle *tektonbundle.Bundle) operands.Operand {
+	newTasks := []pipeline.Task{}
+	for _, task := range bundle.Tasks {
```

Comment:
**future PR** - code style - extract this loop to a function, and then call it here 4 times.

Reply:
Hi, do you mean to:
```
func New(bundle *tektonbundle.Bundle) operands.Operand {
	bundle.Tasks = filterTasks(bundle.Tasks)
	bundle.ServiceAccounts = filterServiceAccounts(bundle.ServiceAccounts)
	bundle.RoleBindings = filterRoleBindings(bundle.RoleBindings)
	bundle.ClusterRoles = filterClusterRoles(bundle.ClusterRoles)

	return &tektonTasks{
		tasks:           bundle.Tasks,
		serviceAccounts: bundle.ServiceAccounts,
		roleBindings:    bundle.RoleBindings,
		clusterRoles:    bundle.ClusterRoles,
	}
}
```
Right?

Reply:
Woah, thank you. I tried to do it via generics, something like this:
```
func filterObjects[T metav1.Object](slice []T, trimSuffix bool) []T {
	var items []T
	for _, item := range slice {
		name := item.GetName()
		if trimSuffix {
			name = strings.TrimSuffix(name, "-task")
		}
		if _, ok := AllowedTasks[name]; ok {
			items = append(items, item)
		}
	}
	return items
}
```
It didn't work and complained with an error:
```
"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1".Task does not implement "k8s.io/apimachinery/pkg/apis/meta/v1".Object (method GetAnnotations has pointer receiver)
```